airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/PostgresReplicationConnection.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.debezium.internals;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.exceptions.ConfigErrorException;
+import io.airbyte.db.factory.DatabaseDriver;
+import io.airbyte.db.jdbc.JdbcUtils;
+import io.debezium.jdbc.JdbcConnection.ResultSetMapper;
+import io.debezium.jdbc.JdbcConnection.StatementFactory;
+import java.sql.*;
+import java.util.*;
+import org.postgresql.util.PSQLException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is a helper for creating replication connections for Postgres
+ */
+public class PostgresReplicationConnection {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PostgresReplicationConnection.class);
+  private static final String ILLEGAL_CONNECTION_ERROR_MESSAGE = "The DB connection is not a valid replication connection";
+  public static final String REPLICATION_PRIVILEGE_ERROR_MESSAGE =
+      "User '%s' does not have enough privileges for CDC replication. Please read the docs and add required privileges.";
+
+  public static Connection createConnection(final JsonNode jdbcConfig) throws SQLException, IllegalStateException {
+    Properties properties = new Properties();
+    properties.setProperty("user", jdbcConfig.has(JdbcUtils.USERNAME_KEY) ? jdbcConfig.get(JdbcUtils.USERNAME_KEY).asText()
+        : null);
+    properties.setProperty("password", jdbcConfig.has(JdbcUtils.PASSWORD_KEY) ? jdbcConfig.get(JdbcUtils.PASSWORD_KEY).asText()
+        : null);
+    properties.setProperty("assumeMinServerVersion", "9.4");
+    properties.setProperty("ApplicationName", "Airbyte Debezium Streaming");
+    properties.setProperty("replication", "database");
+    properties.setProperty("preferQueryMode", "simple"); // replication protocol only supports simple query mode
+
+    LOGGER.info("Creating a CDC connection.");
+    final String jdbcUrl = jdbcConfig.has(JdbcUtils.JDBC_URL_KEY) ? jdbcConfig.get(JdbcUtils.JDBC_URL_KEY).asText()
+        : String.format(DatabaseDriver.POSTGRESQL.getUrlFormatString(),
```

Comment:
I found a bug out here if a connection only accepts SSL connection then we need to add the attributes related to SSL connection while constructing the URL format. Here we are not adding em. It will cause the check to fail if a server only accepts SSL connection. I tested this with a cloud sql instance which only accepts SSL connections